connectivity/FEATURE_BLE/source/generic/GapImpl.cpp

```diff
@@ -2531,6 +2602,36 @@ ble_error_t Gap::stopAdvertising(advertising_handle_t handle)
 
     return status;
 }
+
+#if BLE_FEATURE_EXTENDED_ADVERTISING
+void Gap::evaluate_advertising_stop()
+{
+    // refresh for address for all connectable advertising sets
+    for (size_t i = 0; i < BLE_GAP_MAX_ADVERTISING_SETS; ++i) {
+        if (_pending_stop_sets.get(i)) {
+            ble_error_t status = _pal_gap.extended_advertising_enable(
+                /* enable */ false,
```

Comment:
Why not pass all the handles in a single call ?

Reply:
code simplicity, do you want me to do it in one go?

Reply:
I think it would make things simpler as there would be just one event queued instead of a list.